### PR TITLE
Added user and student endpoints.

### PR DESCRIPTION
**Pull request by kochie**
Added the API routes needed for students, should all work as advertised.

### CODE REVIEW

**Comment by erfanio** [comment]
lol the commas on `Student.js:5` and `Student.js:8` cause syntax error 😂

File: app/validators/User.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const Joi = require('joi');
+
+exports.payload = {
+  firstName: Joi.string(),
+  lastName: Joi.string(),
+  preferredName: Joi.string(),
+  email: Joi.string().email(),
+  dateOfBirth: Joi.date(),
+  gender: Joi.string().uri(),
```
Comment (erfanio) [inline app/validators/User.js]: I'm confused... why is gender a uri?

Comment (hugomd) [inline app/validators/User.js]: ^

Comment (erfanio) [inline app/validators/User.js]: that was fast

**Pull request merged**